examples/src/croncat.rs

```diff
@@ -0,0 +1,194 @@
+    let counter_contract = worker.dev_deploy(COUNTER_CONTRACT).await?;
+
+    // deploy the manager contract so we can schedule tasks via our agents.
+    let manager_contract = worker.dev_deploy(MANAGER_CONTRACT).await?;
+    manager_contract.call(&worker, "new").transact().await?;
+
+    // Create a root croncat account with agent subaccounts to schedule tasks.
+    let croncat = worker.dev_create_account().await?;
+
+    // This will setup a task to call into the counter contract, with a cadence of 1 hour.
+    println!("Creating task for `counter.increment`");
+    let outcome = croncat
+        .call(&worker, manager_contract.id(), "create_task")
+        .args_json(json!({
+            "contract_id": counter_contract.id(),
+            "function_id": "increment",
+            "cadence": "*/1 * * * * *"
+        }))?
+        .max_gas()
+        .deposit(parse_near!("1 N"))
+        .transact()
+        .await?;
+    println!("-- outcome: {:#?}\n", outcome);
+
+    // Let's create an agent that will eventually execute the above task and get rewards
+    // for executing it:
+    let agent_1 = croncat
+        .create_subaccount(&worker, "agent_1")
+        .initial_balance(parse_near!("10 N"))
+        .transact()
+        .await?
+        .into_result()?;
+
+    // Now with all the above setup complete, we can now have the agent run our task:
+    run_scheduled_tasks(&worker, &manager_contract, &agent_1).await?;
+
+    Ok(())
+}
+
+/// This function will schedule a particular task (`counter.increment`) and a single agent
+/// will run that task to eventually get rewards.
+pub async fn run_scheduled_tasks(
+    worker: &Worker<Sandbox>,
+    contract: &Contract,
+    agent: &Account,
+) -> anyhow::Result<()> {
+    // Register the agent to eventually execute the task
+    let outcome = agent
+        .call(&worker, contract.id(), "register_agent")
+        .args_json(json!({}))?
+        .deposit(parse_near!("0.00226 N"))
+        .transact()
+        .await?;
+    println!("Registering agent outcome: {:#?}\n", outcome);
+
+    // Check the right agent was registered correctly:
+    let registered_agent = contract
+        .call(&worker, "get_agent")
+        .args_json(json!({ "account_id": agent.id() }))?
+        .view()
+        .await?
+        .json::<Option<Agent>>()?
+        .unwrap();
+    println!("Registered agent details: {:#?}", registered_agent);
+    assert_eq!(registered_agent.status, AgentStatus::Active);
+    assert_eq!(&registered_agent.payable_account_id, agent.id());
+
+    // Advance 4500 blocks in the chain. 1 block takes approx 1.5 seconds to be produced, but we
+    // don't actually wait that long since we are time travelling to the future via `fast_forward`!
+    // After this `fast_forward` call, we should be ahead by about an hour, and it is expected for
+    // our agents to be able to execute the task we scheduled.
+    println!("Waiting until next slot occurs...");
+    worker.fast_forward(4500).await?;
```

Comment:
Looks awesome! Would love to see it maybe do more forwarding - we have tasks that happen every hour for example, forward a few times proves out the agent model

Reply:
done!